src/resdk/query.py

```diff
@@ -184,8 +184,25 @@ def _dehydrate_resources(self, obj):
     def _add_filter(self, filter_: dict):
         """Add filtering parameters."""
         for key, value in filter_.items():
-            # 'sample' is called 'entity' in the backend.
-            key = key.replace("sample", "entity")
+            # Make best-effort to rename fields to server fields.
+            resource = self.resource
+            filter_parts = key.split("__")
+            for part_num in range(len(filter_parts)):
+                resource_fields = resource._find_fields()
+                part = filter_parts[part_num]
+                # Bail out, we lost track of the book keeping.
+                if part not in resource_fields:
+                    break
+                field = resource_fields[part]
+                if field.server_field != part:
+                    filter_parts[part_num] = resource_fields[part].server_field
```

Comment:
Could just replace the `resource_fields[part].server_field` with `field.server_field`.

Reply:
Sure.